File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-ird-5-4/CashPriceMethod.cpp
// CashPriceMethod.cpp 
#include "CashPriceMethod.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CashPriceMethod::CashPriceMethod(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //cashSettlementReferenceBanksNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* cashSettlementReferenceBanksNode = xmlNode->FirstChildElement("cashSettlementReferenceBanks");

   if(cashSettlementReferenceBanksNode){cashSettlementReferenceBanksIsNull_ = false;}
   else{cashSettlementReferenceBanksIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- cashSettlementReferenceBanksNode , address : " << cashSettlementReferenceBanksNode << std::endl;
   #endif
   if(cashSettlementReferenceBanksNode)
   {
      if(cashSettlementReferenceBanksNode->Attribute("href") || cashSettlementReferenceBanksNode->Attribute("id"))
      {
          if(cashSettlementReferenceBanksNode->Attribute("id"))
          {
             cashSettlementReferenceBanksIDRef_ = cashSettlementReferenceBanksNode->Attribute("id");
             cashSettlementReferenceBanks_ = boost::shared_ptr<CashSettlementReferenceBanks>(new CashSettlementReferenceBanks(cashSettlementReferenceBanksNode));
             cashSettlementReferenceBanks_->setID(cashSettlementReferenceBanksIDRef_);
             IDManager::instance().setID(cashSettlementReferenceBanksIDRef_,cashSettlementReferenceBanks_);
          }
          else if(cashSettlementReferenceBanksNode->Attribute("href")) { cashSettlementReferenceBanksIDRef_ = cashSettlementReferenceBanksNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { cashSettlementReferenceBanks_ = boost::shared_ptr<CashSettlementReferenceBanks>(new CashSettlementReferenceBanks(cashSettlementReferenceBanksNode));}
   }

   //cashSettlementCurrencyNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* cashSettlementCurrencyNode = xmlNode->FirstChildElement("cashSettlementCurrency");

   if(cashSettlementCurrencyNode){cashSettlementCurrencyIsNull_ = false;}
   else{cashSettlementCurrencyIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- cashSettlementCurrencyNode , address : " << cashSettlementCurrencyNode << std::endl;
   #endif
   if(cashSettlementCurrencyNode)
   {
      if(cashSettlementCurrencyNode->Attribute("href") || cashSettlementCurrencyNode->Attribute("id"))
      {
          if(cashSettlementCurrencyNode->Attribute("id"))
          {
             cashSettlementCurrencyIDRef_ = cashSettlementCurrencyNode->Attribute("id");
             cashSettlementCurrency_ = boost::shared_ptr<Currency>(new Currency(cashSettlementCurrencyNode));
             cashSettlementCurrency_->setID(cashSettlementCurrencyIDRef_);
             IDManager::instance().setID(cashSettlementCurrencyIDRef_,cashSettlementCurrency_);
          }
          else if(cashSettlementCurrencyNode->Attribute("href")) { cashSettlementCurrencyIDRef_ = cashSettlementCurrencyNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { cashSettlementCurrency_ = boost::shared_ptr<Currency>(new Currency(cashSettlementCurrencyNode));}
   }

   //quotationRateTypeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* quotationRateTypeNode = xmlNode->FirstChildElement("quotationRateType");

   if(quotationRateTypeNode){quotationRateTypeIsNull_ = false;}
   else{quotationRateTypeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- quotationRateTypeNode , address : " << quotationRateTypeNode << std::endl;
   #endif
   if(quotationRateTypeNode)
   {
      if(quotationRateTypeNode->Attribute("href") || quotationRateTypeNode->Attribute("id"))
      {
          if(quotationRateTypeNode->Attribute("id"))
          {
             quotationRateTypeIDRef_ = quotationRateTypeNode->Attribute("id");
             quotationRateType_ = boost::shared_ptr<QuotationRateTypeEnum>(new QuotationRateTypeEnum(quotationRateTypeNode));
             quotationRateType_->setID(quotationRateTypeIDRef_);
             IDManager::instance().setID(quotationRateTypeIDRef_,quotationRateType_);
          }
          else if(quotationRateTypeNode->Attribute("href")) { quotationRateTypeIDRef_ = quotationRateTypeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { quotationRateType_ = boost::shared_ptr<QuotationRateTypeEnum>(new QuotationRateTypeEnum(quotationRateTypeNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CashSettlementReferenceBanks> CashPriceMethod::getCashSettlementReferenceBanks()
{
   if(!this->cashSettlementReferenceBanksIsNull_){
        if(cashSettlementReferenceBanksIDRef_ != ""){
             return boost::shared_static_cast<CashSettlementReferenceBanks>(IDManager::instance().getID(cashSettlementReferenceBanksIDRef_));
        }else{
             return this->cashSettlementReferenceBanks_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CashSettlementReferenceBanks>();
   }
}
boost::shared_ptr<Currency> CashPriceMethod::getCashSettlementCurrency()
{
   if(!this->cashSettlementCurrencyIsNull_){
        if(cashSettlementCurrencyIDRef_ != ""){
             return boost::shared_static_cast<Currency>(IDManager::instance().getID(cashSettlementCurrencyIDRef_));
        }else{
             return this->cashSettlementCurrency_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Currency>();
   }
}
boost::shared_ptr<QuotationRateTypeEnum> CashPriceMethod::getQuotationRateType()
{
   if(!this->quotationRateTypeIsNull_){
        if(quotationRateTypeIDRef_ != ""){
             return boost::shared_static_cast<QuotationRateTypeEnum>(IDManager::instance().getID(quotationRateTypeIDRef_));
        }else{
             return this->quotationRateType_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<QuotationRateTypeEnum>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-ird-5-4/CashPriceMethod.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CashPriceMethod : ISerialized
    {
        public CashPriceMethod(XmlNode xmlNode)
        {
            XmlNode cashSettlementReferenceBanksNode = xmlNode.SelectSingleNode("cashSettlementReferenceBanks");
            
            if (cashSettlementReferenceBanksNode != null)
            {
                if (cashSettlementReferenceBanksNode.Attributes["href"] != null || cashSettlementReferenceBanksNode.Attributes["id"] != null) 
                {
                    if (cashSettlementReferenceBanksNode.Attributes["id"] != null) 
                    {
                        cashSettlementReferenceBanksIDRef_ = cashSettlementReferenceBanksNode.Attributes["id"].Value;
                        CashSettlementReferenceBanks ob = new CashSettlementReferenceBanks(cashSettlementReferenceBanksNode);
                        IDManager.SetID(cashSettlementReferenceBanksIDRef_, ob);
                    }
                    else if (cashSettlementReferenceBanksNode.Attributes["href"] != null)
                    {
                        cashSettlementReferenceBanksIDRef_ = cashSettlementReferenceBanksNode.Attributes["href"].Value;
                    }
                    else
                    {
                        cashSettlementReferenceBanks_ = new CashSettlementReferenceBanks(cashSettlementReferenceBanksNode);
                    }
                }
                else
                {
                    cashSettlementReferenceBanks_ = new CashSettlementReferenceBanks(cashSettlementReferenceBanksNode);
                }
            }
            
        
            XmlNode cashSettlementCurrencyNode = xmlNode.SelectSingleNode("cashSettlementCurrency");
            
            if (cashSettlementCurrencyNode != null)
            {
                if (cashSettlementCurrencyNode.Attributes["href"] != null || cashSettlementCurrencyNode.Attributes["id"] != null) 
                {
                    if (cashSettlementCurrencyNode.Attributes["id"] != null) 
                    {
                        cashSettlementCurrencyIDRef_ = cashSettlementCurrencyNode.Attributes["id"].Value;
                        Currency ob = new Currency(cashSettlementCurrencyNode);
                        IDManager.SetID(cashSettlementCurrencyIDRef_, ob);
                    }
                    else if (cashSettlementCurrencyNode.Attributes["href"] != null)
                    {
                        cashSettlementCurrencyIDRef_ = cashSettlementCurrencyNode.Attributes["href"].Value;
                    }
                    else
                    {
                        cashSettlementCurrency_ = new Currency(cashSettlementCurrencyNode);
                    }
                }
                else
                {
                    cashSettlementCurrency_ = new Currency(cashSettlementCurrencyNode);
                }
            }
            
        
            XmlNode quotationRateTypeNode = xmlNode.SelectSingleNode("quotationRateType");
            
            if (quotationRateTypeNode != null)
            {
                if (quotationRateTypeNode.Attributes["href"] != null || quotationRateTypeNode.Attributes["id"] != null) 
                {
                    if (quotationRateTypeNode.Attributes["id"] != null) 
                    {
                        quotationRateTypeIDRef_ = quotationRateTypeNode.Attributes["id"].Value;
                        QuotationRateTypeEnum ob = new QuotationRateTypeEnum(quotationRateTypeNode);
                        IDManager.SetID(quotationRateTypeIDRef_, ob);
                    }
                    else if (quotationRateTypeNode.Attributes["href"] != null)
                    {
                        quotationRateTypeIDRef_ = quotationRateTypeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        quotationRateType_ = new QuotationRateTypeEnum(quotationRateTypeNode);
                    }
                }
                else
                {
                    quotationRateType_ = new QuotationRateTypeEnum(quotationRateTypeNode);
                }
            }
            
        
        }
        
    
        #region CashSettlementReferenceBanks_
        private CashSettlementReferenceBanks cashSettlementReferenceBanks_;
        public CashSettlementReferenceBanks CashSettlementReferenceBanks_
        {
            get
            {
                if (this.cashSettlementReferenceBanks_ != null)
                {
                    return this.cashSettlementReferenceBanks_; 
                }
                else if (this.cashSettlementReferenceBanksIDRef_ != null)
                {
                    cashSettlementReferenceBanks_ = IDManager.getID(cashSettlementReferenceBanksIDRef_) as CashSettlementReferenceBanks;
                    return this.cashSettlementReferenceBanks_; 
                }
                else
                {
                      return this.cashSettlementReferenceBanks_; 
                }
            }
            set
            {
                if (this.cashSettlementReferenceBanks_ != value)
                {
                    this.cashSettlementReferenceBanks_ = value;
                }
            }
        }
        #endregion
        
        public string cashSettlementReferenceBanksIDRef_ { get; set; }
        #region CashSettlementCurrency_
        private Currency cashSettlementCurrency_;
        public Currency CashSettlementCurrency_
        {
            get
            {
                if (this.cashSettlementCurrency_ != null)
                {
                    return this.cashSettlementCurrency_; 
                }
                else if (this.cashSettlementCurrencyIDRef_ != null)
                {
                    cashSettlementCurrency_ = IDManager.getID(cashSettlementCurrencyIDRef_) as Currency;
                    return this.cashSettlementCurrency_; 
                }
                else
                {
                      return this.cashSettlementCurrency_; 
                }
            }
            set
            {
                if (this.cashSettlementCurrency_ != value)
                {
                    this.cashSettlementCurrency_ = value;
                }
            }
        }
        #endregion
        
        public string cashSettlementCurrencyIDRef_ { get; set; }
        #region QuotationRateType_
        private QuotationRateTypeEnum quotationRateType_;
        public QuotationRateTypeEnum QuotationRateType_
        {
            get
            {
                if (this.quotationRateType_ != null)
                {
                    return this.quotationRateType_; 
                }
                else if (this.quotationRateTypeIDRef_ != null)
                {
                    quotationRateType_ = IDManager.getID(quotationRateTypeIDRef_) as QuotationRateTypeEnum;
                    return this.quotationRateType_; 
                }
                else
                {
                      return this.quotationRateType_; 
                }
            }
            set
            {
                if (this.quotationRateType_ != value)
                {
                    this.quotationRateType_ = value;
                }
            }
        }
        #endregion
        
        public string quotationRateTypeIDRef_ { get; set; }
        
    
        
    
    }
    
}

